decode.go

```diff
@@ -49,6 +49,11 @@ func Decode(buf []byte, structPtr interface{}) error {
 // DecodeWithConstructors is like Decode, but you can pass a map of
 // constructors with which to instantiate interface types.
 func DecodeWithConstructors(buf []byte, structPtr interface{}, cons Constructors) error {
+	defer func() {
+		if e := recover(); e != nil {
+			fmt.Println(e.(string))
```

Comment:
Sorry, this won't work. Library code should return errors, not print them. The caller might need to get that error and decide something, or might want to print it to a logfile instead of to stdout.